WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -719,22 +772,119 @@ private extension EditableOrderViewModel {
+
+            if let input = createUpdateProductInput(item: item, quantity: 0) {
+                inputsToBeRemoved.append(input)
+            }
+
+            analytics.track(event: WooAnalyticsEvent.Orders.orderProductRemove(flow: flow.analyticsFlow))
+        }
+
+        return inputsToBeRemoved
+
+    }
+
+    /// Adds, or removes multiple products from an Order
+    ///
+    func syncOrderItems(products: [Product?], variations: [ProductVariation?]) {
+        // We need to send all OrderSyncProductInput in one call to the RemoteOrderSynchronizer, both additions and deletions
+        // otherwise may ignore the subsequent values that are sent
+        let addedItemsToSync = productInputAdditionsToSync(products: products, variations: variations)
+        let removedItemsToSync = productInputDeletionsToSync(products: products, variations: variations)
+        orderSynchronizer.setProducts.send(addedItemsToSync + removedItemsToSync)
+
+        if addedItemsToSync.isNotEmpty {
+            analytics.track(event: WooAnalyticsEvent.Orders.orderProductAdd(flow: flow.analyticsFlow))
+        }
+
+        if removedItemsToSync.isNotEmpty {
+            analytics.track(event: WooAnalyticsEvent.Orders.orderProductRemove(flow: flow.analyticsFlow))
+        }
+    }
+
     /// Adds a selected product (from the product list) to the order.
     ///
+    // TODO: This method needs to be renamed
+    // to reflect that adds products to Order for single selection, but only selects/unselects for multi-selection
     func addProductToOrder(_ product: Product) {
         // Needed because `allProducts` is only updated at start, so product from new pages are not synced.
         if !allProducts.contains(product) {
             allProducts.append(product)
         }
 
-        let input = OrderSyncProductInput(product: .product(product), quantity: 1)
-        orderSynchronizer.setProduct.send(input)
-
-        analytics.track(event: WooAnalyticsEvent.Orders.orderProductAdd(flow: flow.analyticsFlow))
+        guard featureFlagService.isFeatureFlagEnabled(.productMultiSelectionM1), isProductMultiSelectionBetaFeatureEnabled else {
+            // Single-selection
+            let input = OrderSyncProductInput(product: .product(product), quantity: 1)
+            orderSynchronizer.setProduct.send(input)
+            analytics.track(event: WooAnalyticsEvent.Orders.orderProductAdd(flow: flow.analyticsFlow))
+            return
+        }
+        // Multi-selection
+        if !selectedProducts.contains(where: { $0?.productID == product.productID }) {
+            selectedProducts.append(product)
+        } else {
+            selectedProducts.removeAll(where: { $0?.productID == product.productID })
+        }
     }
 
     /// Adds a selected product variation (from the product list) to the order.
     ///
+    // TODO: This method needs to be renamed
+    // to reflect that adds variations to Order for single selection, but only selects/unselects for multi-selection
```

Comment:
Same as with `addProductToOrder`, leaving this for a future PR to not add further changes to this one.

Reply:
Tracked on eac327e7ee1b0102e264190e345e7e0f4c40efbb